automata/fa/dfa.py

```diff
@@ -39,9 +41,11 @@
 DFAPathT = Mapping[DFASymbolT, DFAStateT]
 DFATransitionsT = Mapping[DFAStateT, DFAPathT]
 
-ExpandStateFn = Callable[[DFAStateT], Iterator[Tuple[DFASymbolT, DFAStateT]]]
+ExpandStateReturnType = Iterator[Tuple[DFASymbolT, DFAStateT]]
+ExpandStateFn = Callable[[DFAStateT], ExpandStateReturnType]
 IsFinalStateFn = Callable[[DFAStateT], bool]
 TargetStateFn = Callable[[DFAStateT], bool]
+BooleanLiteral = Literal[True, False]
```

Comment:
@eliotwrobson @EduardoGoulart1 I believe `BooleanLiteral` is unnecessary. You can simply use `bool` as the type hint (e.g. `lhs_relevant: bool` instead of `lhs_relevant: BooleanLiteral`).